File: array/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {

        // init
        int count = 0;

        // move valid elements to the front
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[count] = nums[i];
                count++;
            }
        }

        // return the number of valid elements
        return count;
    }
};

File: array/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """

        # init
        count = 0

        # move valid elements to the front
        for i in range(len(nums)):
            if nums[i] != val:
                nums[count] = nums[i]
                count += 1

        # return the number of valid elements
        return count